src/Movie.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+const Movie = props => {
+  function rocketFavText(id) {
+    // Toggle the text of RocketFave section if that movie is a savedMovie or not
+    const savedMovieIdArr = (props.savedMovies.map(savedMovie => {
+      return savedMovie.id;
+    }));
+    if (savedMovieIdArr.indexOf(id) >= 0) {
+      return 'Remove RocketFave';
+    } else {
+      return 'Add to RocketFaves';
+    }
+  }
+  function checkLength() {
+    // If there are movies in the App.js movie state, then map over them
+    if (props.movies.length) {
+      return (
+        props.movies.map((movieResult) => {
```

Comment:
This map should really be taking place in your `MovieList` component, and this component should be responsible for rendering a single _movie_